image_handler.py

```
@@ -0,0 +1,328 @@
+    try:
+        image_bytes = await download_image(url)
+        if not image_bytes:
+            return None
+        
+        # Compress image if requested
+        was_compressed = False
+        if compress:
+            try:
+                image_bytes = compress_image(image_bytes, max_size=max_size, quality=quality)
+                was_compressed = True
+            except Exception as e:
```

Ruff (0.14.2)

Do not catch blind exception: `Exception`

(BLE001)